components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/UserSessionStore.java

```diff
@@ -871,4 +872,21 @@ public int getActiveSessionCount(String tenantDomain) throws UserSessionExceptio
         }
         return activeSessionCount;
     }
+
+    /**
+     * Returns the user id of the federated user.
+     *
+     * @param subjectIdentifier - Subject Identifier of the federated user.
+     * @param tenantId          - Id of the tenant domain.
+     * @param idPId             - Id of the identity provider.
+     * @return userId - User Id of the federated user.
+     * @throws UserSessionException
+     */
+    public String getFederatedUserId(String subjectIdentifier, int tenantId, int idPId)
```

Comment:
What would be the tenantId here?
Is it the IDP tenant ID?

Reply:
[Here](https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java#L517 ) we get the tenant domain from the request and set it into the context and [here](https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java#L226) we set the tenant domain to the user from the context. From [here](https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java#L638) we get the tenant domain from the user and get the tenant id for that and [here](https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java#L656) we store the federated user into the user store.